tests/test_ensemble_tools.py

```diff
@@ -0,0 +1,239 @@
+"""Tests for the ensemble_tools utilities."""
+
+import itertools
+import re
+import shutil
+from pathlib import Path
+from xml.etree import ElementTree
+
+import pytest
+
+from FabNESO.ensemble_tools import create_dict_sweep, create_dir_tree, edit_parameters
+
+
+def test_edit_parameters(tmp_path: Path) -> None:
+    """Test the edit_parameters method of ensemble_tools."""
+    parameter_to_test = "particle_initial_velocity"
+    parameter_test_value = 10.0
+    temp_conditions_path = tmp_path / "conditions.xml"
+    shutil.copyfile(
+        Path(__file__).parents[1] / "config_files" / "two_stream" / "conditions.xml",
+        temp_conditions_path,
+    )
+
+    edit_parameters(temp_conditions_path, {parameter_to_test: parameter_test_value})
+
+    assert _check_parameter_in_conditions(
+        temp_conditions_path, parameter_to_test, parameter_test_value
+    ) == (1, 0)
```

Comment:
```suggestion
    n_equal_in_value, n_different_in_value = _check_parameter_in_conditions(
        temp_conditions_path, parameter_to_test, parameter_test_value
    )
    assert n_equal_in_value == 1 and n_different_in_value == 0
```
Good idea to have the check function output two different values here. To help make the test a bit more self documenting in case of a failure (as Pytest will show this assert statement on a failure) it might be good to attach names to the outputs which make it clearer why we expect values of 1 and 0.